src/domUpdates.js

```diff
@@ -22,17 +71,31 @@ const tagContainer = document.querySelector("#tagContainer");
 const saveRecipeButton = document.querySelector(".save-button");
 
 let currentUser;
+let users = [];
+let ingredients = [];
+let recipes = [];
 
 function initialize() {
-  generateRecipeCards();
-  populateAllRecipesPage();
-  // showFullRecipe();
-  // displayUserRecipes(userName)
-  currentUser = getRandomUser();
-  console.log("Current user: ", currentUser);
+  Promise.all([fetchUsers(), fetchIngredients(), fetchRecipes()])
+    .then(([fetchedUsers, fetchedIngredients, fetchedRecipes]) => {
+      users = fetchedUsers;
+      ingredients = fetchedIngredients;
+      recipes = fetchedRecipes;
+      
+      currentUser = getRandomUser(users);
+      populateAllRecipesPage(recipes);
+      generateRecipeCards(recipes);
```

Comment:
By the way, in response to "why it's breaking", I believe this is what is breaking it.  You're passing recipes to `populateAllRecipesPage` assuming that `recipes` is an array.